clients/src/test/java/org/apache/kafka/common/security/authenticator/SaslAuthenticatorTest.java

```diff
@@ -384,7 +414,16 @@ public void testUnauthenticatedApiVersionsRequestOverSsl() throws Exception {
      * {@link SaslServerAuthenticator} of the server prior to client authentication.
      */
     @Test
-    public void testApiVersionsRequestWithUnsupportedVersion() throws Exception {
+    public void testApiVersionsRequestWithUnsupportedVersionHandshakeVersion0() throws Exception {
+        verifyApiVersionsRequestWithUnsupportedVersion((short) 0);
+    }
+
+    @Test
+    public void testApiVersionsRequestWithUnsupportedVersionHandshakeVersion1() throws Exception {
```

Comment:
Is it worth testing this case? It seems that the ApiVersion part is covered in testApiVersionsRequestWithUnsupportedVersionHandshakeVersion0 and the SaslHandshake part is covered in testUnauthenticatedApiVersionsRequestOverSslHandshakeVersion0 and testUnauthenticatedApiVersionsRequestOverSslHandshakeVersion1?

Reply:
Removed unnecessary test, retaining a single test with latest handshake request version.